### DISCUSSION TITLE
Route function runs for every href on page

### DISCUSSION THREAD

**Discussion by john-cashin** [vikejs/vike#1073] [Discussion] [answered]
I've a SSR single page app with client side routing. From reading the docs it mentions every route function runs on navigation, I assume thats every defined *.page.route.js?. But it's hitting the `index.page.route.js` or `_default.page.route.js` for each href on the page be it 1 or 100. Is this the intended behaviour? 

Disabling clientside seems to correct only fire the route functions once.

**Comment by brillout** [comment]
If you enabled [viewport link prefetching](https://vite-plugin-ssr.com/clientRouting#link-prefetching) then yes that's the intended behavior. If you didn't then let me know as it would be unexpected and should be fixed.

**Comment by john-cashin** [comment]
I've set prefetching to false, which shouldn't be active during development as mentioned in the docs. But I'm still seeing the issue occur locally. 

My basic `onBeforerRoute`. I get the hit on SSR, then on client load It logs for each href present. On user navigation it logs the new route and any new hrefs on the page. One of these logs contains the `documentProps` of the new page, which I thought I could then use as a flag to but its `undefined` when trying to access. I'd assume the render has also already occured if I have the `docmentProps` in the `pageContext`


```
  function onBeforeRoute(pageContext) {
    const { documentProps } = pageContext // shows in log in dev tools but is always defined
  
    if (documentProps.title || import.meta.env.SSR) {
      if (import.meta.env.VITE_BRAND === 'brand') {
        //
        //
      }
    }
    return {
      pageContext: {
        urlOriginal: pageContext.urlOriginal
      }
    }
  }
```

**Reply by brillout** [reply]
Indeed, I could reproduce. Let me have a look.

**Reply by brillout** [reply, marked as answer]
Fix (575d48c96) released as `0.4.136-commit-575d48c`. I believe it should resolve your issue; let me know whether it does.

Btw. in case your company is up for it: https://github.com/sponsors/brillout.
